editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-props-utils.ts

```diff
@@ -38,39 +38,36 @@ export function applyPropsParamToPassedProps(
       output[paramName] = getParamValue(value, boundParam.defaultExpression)
     } else if (isDestructuredObject(boundParam)) {
       if (typeof value === 'object' && !Array.isArray(value) && value !== null) {
+        const valueAsRecord: Record<string, unknown> = { ...value }
         let remainingValues = { ...value } as Record<string, unknown>
-        let remainingKeys = Object.keys(remainingValues)
-        boundParam.parts.forEach((part) => {
+        for (const part of boundParam.parts) {
           const { propertyName, param } = part
-          if (propertyName != null) {
-            // e.g. `{ prop: renamedProp }` or `{ prop: { /* further destructuring */ } }`
-            // Can't spread if we have a property name
-            const innerValue = remainingValues[propertyName]
-            applyBoundParamToOutput(innerValue, param.boundParam)
-            remainingKeys = remainingKeys.filter((k) => k !== propertyName)
-            delete remainingValues[propertyName]
-          } else {
+          if (propertyName == null) {
             const { dotDotDotToken: spread, boundParam: innerBoundParam } = param
             if (isRegularParam(innerBoundParam)) {
               // e.g. `{ prop }` or `{ ...remainingProps }`
               const { paramName } = innerBoundParam
               if (spread) {
                 output[paramName] = remainingValues
-                remainingKeys = []
                 remainingValues = {}
               } else {
                 output[paramName] = getParamValue(
-                  remainingValues[paramName],
+                  valueAsRecord[paramName],
                   innerBoundParam.defaultExpression,
                 )
-                remainingKeys = remainingKeys.filter((k) => k !== paramName)
                 delete remainingValues[paramName]
               }
             }
+          } else {
+            // e.g. `{ prop: renamedProp }` or `{ prop: { /* further destructuring */ } }`
+            // Can't spread if we have a property name
+            const innerValue = valueAsRecord[propertyName]
```

Comment:
can you explain what was the actual fix here? I see you introduced valueAsRecord which seems to be the same as remainingValues except we don't delete entries from it in the loop. but I don't see how did that fix the behavior

Reply:
Previously it was using `remainingValues` and deleting from that object, but it's totally legitimate to refer to a field more than once. Like you could destructure `style` but also refer to `style` itself in the same case. As `remainingValues` was getting trimmed this meant that every use of a field after the first was getting an `undefined`.

Reply:
ahaa I see

Reply:
Oh wow, nice catch!